### DISCUSSION TITLE
Benchmarking - Add effective mounted volume size and families benchmark times to asset scan

### DISCUSSION THREAD

**Discussion by fishkerez** [openclarity/vmclarity#465] [Ideas]
We would like to  be able to estimate the time and cost of scans before the scans are running, in order to give the user a better understanding of what will be the cost and resource utilization of the scan before he starts scanning.
To be able to estimate the time and cost of scans, it is necessary to gather actual scanned filesystem size and time benchmarks per plugged scanner. This information can then be added to the assetScan that API object that is reported the the backend. 

**Time measurement :**
add new benchmark field to asset scan that will contain start and end time for each family, and also a general scan time.

There are 2 options for measuring time:
1. measure time per scanner - require more infrastructure change (also in kubeclarity). In addition,  scanners are running in parallel, so the measurement will be only for the scanner that took the most time.
2. measure time per family - more simple, we can hook into the current export logic in the CLI manager, and report the times as part of the InProgress and Done reporting of the AssetScan. can compare scan configs.

**Size calculations:**
need to report size of supported input types only:
for directories, need to calculate the total dir size
for images, calculate image size
for exploits, sboms - not relevant

The size will be reported as the total size of all the inputs to the family scan, and should also include the mounted volumes size.

**Comment by fishkerez** [comment]
API:
https://github.com/openclarity/vmclarity/pull/473

**Reply by akpsgit** [reply]
Cool! please check the comments on the draft

**Reply by akpsgit** [reply]
Looks great
